tests/testthat/test-problems.R

```diff
@@ -19,42 +15,34 @@ test_that("problems works for multiple files", {
   writeLines("x,y\n1,2\n1,1.x\n2,2", out1)
   writeLines("x,y\n3.x,4\n1,2\n2,2", out2)
 
-  expect_warning(
+  expect_snapshot({
     x <- vroom(c(out1, out2), delim = ",", col_types = "dd", altrep=F)
-  )
-  probs <- problems(x)
+    problems(x)
+  })
 
-  expect_equal(probs$row, c(3, 2))
-  expect_equal(probs$col, c(2, 1))
-  expect_equal(probs$expected, c("a double", "a double"))
-  expect_equal(probs$actual, c("1.x", "3.x"))
-  expect_equal(basename(probs$file), basename(c(out1, out2)))
 })
 
 test_that("problems with number of columns works for single files", {
-  expect_warning(probs3 <- problems(vroom(I("x,y,z\n1,2\n"), col_names = TRUE, col_types = "ddd", altrep = FALSE)))
-  expect_equal(probs3$row, 2)
-  expect_equal(probs3$col, 2)
-  expect_equal(probs3$expected, "3 columns")
-  expect_equal(probs3$actual, "2 columns")
-
-  expect_warning(probs3 <- problems(vroom(I("x,y,z\n1,2\n"), col_names = FALSE, col_types = "ddd", altrep = FALSE)))
-  expect_equal(probs3$row[[4]], 2)
-  expect_equal(probs3$col[[4]], 2)
-  expect_equal(probs3$expected[[4]], "3 columns")
-  expect_equal(probs3$actual[[4]], "2 columns")
-
-  expect_warning(probs4 <- problems(vroom(I("x,y\n1,2,3,4\n"), col_names = TRUE, col_types = "dd", altrep = FALSE)))
-  expect_equal(probs4$row[[2]], 2)
-  expect_equal(probs4$col[[2]], 4)
-  expect_equal(probs4$expected[[2]], "2 columns")
-  expect_equal(probs4$actual[[2]], "4 columns")
-
-  expect_warning(probs2 <- problems(vroom(I("x,y\n1,2,3,4\n"), col_names = FALSE, col_types = "dd", altrep = FALSE)))
-  expect_equal(probs2$row[[4]], 2)
-  expect_equal(probs2$col[[4]], 4)
-  expect_equal(probs2$expected[[4]], "2 columns")
-  expect_equal(probs2$actual[[4]], "4 columns")
+
+  expect_snapshot({
+    v1 <- vroom(I("x,y,z\n1,2\n"), col_names = TRUE, col_types = "ddd", altrep = FALSE)
+    problems(v1)
+  })
+
+  expect_snapshot({
+    v2 <- vroom(I("x,y,z\n1,2\n"), col_names = FALSE, col_types = "ddd", altrep = FALSE)
+    problems(v2)
+  })
+
+  expect_snapshot({
+    v3 <- vroom(I("x,y\n1,2,3,4\n"), col_names = TRUE, col_types = "dd", altrep = FALSE)
+    problems(v3)
+  })
+
+ expect_snapshot({
+    v4 <- vroom(I("x,y\n1,2,3,4\n"), col_names = FALSE, col_types = "dd", altrep = FALSE)
+    problems(v4)
+  })
 })
 
 test_that("parsing problems are shown for all datatypes", {
```

Comment:
I'm still thinking about how to use snapshots here. My first take was replacing the `expect_warning()` and `expect_equal()` with `expect_snapshot()`but this felt repetitive because of the looping.